tests/unit/utils/test_validations.py

```diff
@@ -0,0 +1,498 @@
+        str_test_variable="test",
+        int_test_variable=8,
+        float_test_variable=3.14,
+        list_test_variable=my_list,
+        test_uri="http://example.com",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 0
+
+
+def test_validate_float_type_incorrect():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting an float,
+    but it was provided with a non float type variable.
+    """
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable="test"
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value, check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_nan_incorrect():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting an float,
+    but it was provided with a float("nan") instead.
+    """
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable=float("nan")
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value but is Not a Number (NaN), check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_infinity_incorrect():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting an float,
+    but it was provided with a float("inf") instead.
+    """
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable=float("inf")
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value but is +-infinity, check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_neg_infinity_incorrect():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting an float,
+    but it was provided with a float("-inf") instead.
+    """
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable=float("-inf")
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value but is +-infinity, check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_correct():
+    """This test will check if the class with the correct argument types will pass the validation"""
```

Comment:
Can you make this a bit more specific?

Reply:
Docstring edited to be more specific and mention what type it concerns.